src/ripple/basics/impl/StringUtilities.cpp

```diff
@@ -31,14 +32,11 @@
 
 namespace ripple {
 
-// NIKB NOTE: This function is only used by strUnHex (std::string const& strSrc)
-// which results in a pointless copy from std::string into std::vector. Should
-// we just scrap this function altogether?
-int strUnHex (std::string& strDst, std::string const& strSrc)
+std::pair<Blob, bool> strUnHex (std::string const& strSrc)
```

Comment:
Passing an iterator in requires knowing the size in advance, or using an output iterator. Its ugly no matter what the viewing angle.